src/main/java/com/google/api/codegen/transformer/py/PythonSamplePrintArgTransformer.java

```diff
@@ -0,0 +1,95 @@
+/* Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer.py;
+
+import static com.google.api.codegen.viewmodel.PrintArgView.ArgSegmentView;
+import static com.google.api.codegen.viewmodel.PrintArgView.TextSegmentView;
+import static com.google.api.codegen.viewmodel.PrintArgView.VariableSegmentView;
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.api.codegen.config.ProtoTypeRef;
+import com.google.api.codegen.config.TypeModel;
+import com.google.api.codegen.transformer.MethodContext;
+import com.google.api.codegen.transformer.OutputTransformer;
+import com.google.api.codegen.viewmodel.OutputView;
+import com.google.api.codegen.viewmodel.PrintArgView;
+import com.google.api.tools.framework.model.ProtoElement;
+import com.google.api.tools.framework.model.ProtoFile;
+import com.google.api.tools.framework.model.TypeRef;
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PythonSamplePrintArgTransformer implements OutputTransformer.PrintArgTransformer {
+
+  /**
+   * Converts the {@link OutputView#VariableView} to a {@link PrintArgView} that knows how to print
+   * proto types nicely. Has effects only when the variable is an enum.
+   *
+   * <p>In python, calling {@code print(x)} or {@code print(''.format(x))} would only get an int
+   * when {@code x} is an enum. To print the enum name of the value, we need to convert the enum
+   * value to it's name.
+   *
+   * <p>When the type of the enum value {@code x} is a top-level enum {@code EnumType}, we can call
+   * {@code print(enums.EnumType(x).name)} to print the name of {@code x}.
+   *
+   * <p>When the type of the enum value {@code x} is a nested enum {@code MessageType.EnumType}, we
+   * can call {@code print(enums.MessageType.EnumType(x).name)} to print the name of {@code x}.
+   */
+  @Override
+  public PrintArgView generatePrintArg(
+      MethodContext context, OutputView.VariableView variableView) {
+    if (variableView.type() != null && variableView.type().isEnum()) {
```

Comment:
I think we should invert this, checking for variable not being enum first. In general, I think it's good practice to handle easy cases earlier.